### ISSUE TITLE
v0.11.0 not playing custom chime sounds, only tts

### ISSUE THREAD

**Issue by jchristianj** [nimroddolev/chime_tts#69]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/nimroddolev/chime_tts/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

After updating from v0.10.2 to v0.11.0 no custom chime sound is played any longer; tts message consists of the spoken text only. Did not change any setting or path. HA log entry is similar "could not play file None", do not remember exactly. 

Preset sounds are played, only custom sounds do not work.

Went back to v0.10.2 and all is fine again.

### Describe the solution you'd like

Custom sounds should be played.

### Describe alternatives you've considered

went back to previous version, all is fine again.

### Additional context

![image](https://github.com/nimroddolev/chime_tts/assets/51285869/f21ad8a8-584a-4cbf-8b77-0c03c20fb344)

**Comment by nimroddolev** [comment]
Hi @jchristianj, I’m sorry the new release has caused issues for you.

Is the screenshot you included from the configuration in the new version? Do the paths to the custom paths appear when you downgraded to the previous version?

There is a fix in a new beta version which might address your issue. Could you please upgrade to it and let me know if it works for you?

To install the new beta version:
1. Open the <a href="https://my.home-assistant.io/redirect/hacs_repository/?owner=nimroddolev&repository=chime_tts&category=integration" title="Chime TTS HACS repository" target="_blank">Chime TTS repository in HACS</a>
1. Click on the `⋮` button in the top right
1. Select the `↻ Redownload` option
1. Enable the `Show beta versions` option
1. Select version `v0.11.1-beta2` from the updated version drop-down list

**Comment by Wummeke** [comment]
Same issue here.
I have a file in /media/sounds called DB-Regio.mp3. This is the service call:

```
service: chime_tts.say
target:
  entity_id:
    - media_player.esp_muse_luxe
data:
  tts_platform: tts.piper
  message: Hallo welt!
  cache: false
  chime_path: /media/sounds/DB-Regio.mp3
```

but the chime is not played. The `tada` chime from the examples works fine.
The error in the log:

```
2024-01-07 21:30:14.385 WARNING (MainThread) [custom_components.chime_tts] Unable to find audio file None
```

I am on version `v0.11.1-beta2` by the way.

**Comment by jchristianj** [comment]
Hi @nimroddolev thanks for your quick response. The screenshot is from the old version v0.10.2 after I reverted back to it. 

However, the suggested beta2 unfortunately does not work. After going back to to v0.10.2 it's working again.

I am attaching two debug logs, one for the beta2 (not working) and one for v0.10.2 directly after going back (working). Both logs show the same script executed, nothing changed except the version in between. 

 If you need any further information please contact me. Thanks!

[old_version_working_home-assistant_chime_tts_2024-01-07T20-33-22.399Z.log](https://github.com/nimroddolev/chime_tts/files/13854879/old_version_working_home-assistant_chime_tts_2024-01-07T20-33-22.399Z.log)
[beta2_not_working_home-assistant_chime_tts_2024-01-07T20-27-51.984Z.log](https://github.com/nimroddolev/chime_tts/files/13854880/beta2_not_working_home-assistant_chime_tts_2024-01-07T20-27-51.984Z.log)

**Comment by Wummeke** [comment]
I noticed it does work when I add the mp3 file as Custom 1 to the settings. However, according to the wiki the way I tried it before should work: https://github.com/nimroddolev/chime_tts/wiki/chime_tts.say#chime_path
I tried to add another mp3 from the same folder (/media/sounds) as Custom 2, but that doesn't work:
`Invalid custom chime path detected` 
I can't edit Custom 1 either, that fails with the same error. (`Invalid custom chime path detected` )

**Comment by nimroddolev** [comment]
Hi @jchristianj  and @Wummeke,

Thank you both for the additional information. Unfortunately from my own testing I can't replicate the problem you are experiencing. I reviewed my changes since the previous version and nothing jumps out as the potential cause. I have some follow up questions and troubleshooting I would like to ask you to kindly work through with me. I appreciate your help and your patience 👍 

@Wummeke: it might prove useful to have access to your debug logs. Could you please provide them? To enable debug logs:
  1. Turn on debug logging:
  ```
  service: logger.set_level
  data:
    custom_components.chime_tts: debug
  ```
  2. Call the service chime_tts.say again.
  3. Check your log messages in Home Assistant:
  ```
  https://{YOUR_HOME_ASSISTANT_ADDRESS}:8123/config/logs?filter=chime_tts
  ```
  4. Click `LOAD FULL LOGS`


Also, you mentioned that:
> it does work when I add the mp3 file as Custom 1 to the settings. However, according to the wiki the way I tried it before should work
1. How did you add the mp3 file as Custom 1?
2. What do you mean by "the way I tried it before"?


@jchristianj  - I see from your debug logs of the service call to v0.11.1-beta2 that the custom chime was not found at the path `/media/sounds/a320_tritone-ding-dong.mp3`. At first I thought perhaps the issue is with the path being relative (not originating from the root folder on the filesystem) but further down in the logs I see the final mp3 is saved successfully with the same root folder.

---

To try to understand the cause of issue I have some basic troubleshooting steps I'd like to ask you both to please try. Please go through them (even if they seem obvious) and let me know what your results are:

1. Make sure to be running Chime TTS v0.11.1-beta2
1. Please test and confirm that using the preset chimes from the drop-down UI work
1. Download one of those preset chime mp3 files ([you can download them from here](https://github.com/nimroddolev/chime_tts/tree/main/custom_components/chime_tts/mp3s)) and save it to your media folder, add it as one of your custom chime slots in the integration's configuration dialog, and try to use it as a custom chime.
1. In the configuration dialog, check that the file path to the custom chime _that won't play_ is **exactly the same** as the actual file path on the filesystem (pay attention to uppercase & lowercase letters, etc.).
1. Are you able to add the same chime file that won't play to another custom chime slot? If so, does it play when selected?
1. Please make a copy of the chime file that won't play and rename it so that it has no special characters, eg: "mychime.mp3" (I noticed that you both have dashes in the chime filenames).

**Comment by jchristianj** [comment]
@nimroddolev I really appreciate your support. I will go through the steps tomorrow and inform about the results.  Briefly reviewing the logs and your troubleshooting steps just now, one thing I noticed: The actual file name of the custom chime is

`A320_tritone-ding-dong.mp3`

In the non-working logs the file is named `a320_tritone-ding-dong.mp3` (with lower-case letter a in the beginning)
In the working logs the file is named `A320_tritone-ding-dong.mp3` (upper-case letter A in the beginning, as it is actually named) Could this already be the culprit?

**Comment by nimroddolev** [comment]
I also noticed the lowercase change when reviewing my code changes, and thought about it being the case, but I tested two custom chimes (both the same mp3 but one named with capitals and the other all lowercase) and they both worked.

However: perhaps your setup is more case sensitive than mine? You can try the same approach and see if there's any difference with lowercase files.

**Comment by jchristianj** [comment]
> However: perhaps your setup is more case sensitive than mine? You can try the same approach and see if there's any difference with lowercase files.

This, somehow, seems to be the case. Simply created a copy of `A320_tritone-ding-dong.mp3` in same location but only with lowercase letters `a320_tritone-ding-dong.mp3`, installed v0.11.0 (non-beta) and now it is working when chime path is set to
`chime_path: /media/sounds/a320_tritone-ding-dong.mp3`

**Comment by nimroddolev** [comment]
Great news!

I just released a new beta version, v0.11.1-beta3, which includes a fix for the lowercase bug. It would be great if you and @Wummeke could download the new version and report back if everything is working again.

**Comment by Wummeke** [comment]
Just tested it with Version v0.11.1-beta3 and got mixed results, which i can't explain.

I have 2 files in /media/sounds:
- DB-Regio.mp3
- DB-Regio_leise.mp3
I've put them there last night, so the were already there when I restarted Home-Assistant after updating chime_tts to v0.11.1-beta3.
See attached log for the results. The gist of it is that everything works fine with DB-Regio.mp3, ~~but not with DB-Regio_leise.mp3. Although the latter is virtually the same file as the first, but renamed (in other words, I'm sure it is a valid MP3 file).~~ Appearently I am an idiot. I have a folder `/media/sounds`, but also a folder ` /config/media/sounds/` I thought I was using the last one, but was actually using the first one, and that folder doesn't contain DB-Regio_leise.mp3. I'll sit in the corner of my room with a donkeyhat on my head for the rest of the evening.

So, I'm all good now with your latest beta version! Thanks for your support and keep up the good work (y)

[home-assistant_2024-01-08T18-36-03.665Z.log](https://github.com/nimroddolev/chime_tts/files/13865137/home-assistant_2024-01-08T18-36-03.665Z.log)
